packages/common/src/templates/HubTemplate.ts

```diff
@@ -0,0 +1,225 @@
+   * @param partialTemplate
+   * @param context
+   */
+  private static applyDefaults(
+    partialTemplate: Partial<IHubTemplate>,
+    context: IArcGISContext
+  ): IHubTemplate {
+    // ensure we have the orgUrlKey
+    if (!partialTemplate.orgUrlKey) {
+      partialTemplate.orgUrlKey = context.portal.urlKey;
+    }
+    // extend the partial over the defaults
+    const pojo = { ...DEFAULT_TEMPLATE, ...partialTemplate } as IHubTemplate;
+    return pojo;
+  }
+
+  /*
+   * Get a specific editor config for the HubTemplate entity.
+   * @param i18nScope
+   * @param type
+   */
+  async getEditorConfig(
+    i18nScope: string,
+    type: TemplateEditorType
+  ): Promise<IEditorConfig> {
+    // delegate to the schema subsystem
+    return getEditorConfig(i18nScope, type, this.entity, this.context);
+  }
+
+  /**
+   * Transform template entity into an editor object
+   * @param editorContext
+   */
+  async toEditor(
+    editorContext: IEntityEditorContext = {},
+    include: string[] = []
+  ): Promise<IHubTemplateEditor> {
+    // 1. optionally enrich entity and cast to editor
+    const editor = include.length
+      ? ((await enrichEntity(
+          cloneObject(this.entity),
+          include,
+          this.context.hubRequestOptions
+        )) as IHubTemplateEditor)
+      : (cloneObject(this.entity) as IHubTemplateEditor);
+
+    // 2. Apply transforms to relevant entity values so they
+    // can be consumed by the editor
+
+    return editor;
+  }
+
+  /**
+   * Transform editor values into a template entity
+   * @param editor
+   */
+  async fromEditor(editor: IHubTemplateEditor): Promise<IHubTemplate> {
+    // Setting the thumbnailCache will ensure that the
+    // thumbnail is updated on the next save
+    if (editor._thumbnail) {
+      if (editor._thumbnail.blob) {
+        this.thumbnailCache = {
+          file: editor._thumbnail.blob,
+          filename: editor._thumbnail.filename,
+          clear: false,
+        };
+      } else {
+        this.thumbnailCache = {
+          clear: true,
+        };
+      }
+    }
+
+    delete editor._thumbnail;
+
+    const entity = editorToTemplate(editor, this.context.portal);
+
+    // save, which will also create
+    this.entity = entity;
+    await this.save();
+
+    return this.entity;
+  }
+
+  /**
+   * Update the instance's internal entity state
+   * @param changes
+   */
+  update(changes: Partial<IHubTemplate>): void {
+    if (this.isDestroyed) {
+      throw new Error("HubTemplate is already destroyed.");
+    }
+
+    this.entity = { ...this.entity, ...changes };
+  }
+
+  /** Save the HubTemplate to the backing store */
+  async save(): Promise<void> {
+    if (this.isDestroyed) {
+      throw new Error("HubTemplate is already destroyed.");
+    }
+
+    // 1. create or update
+    this.entity = this.entity.id
+      ? await updateTemplate(this.entity, this.context.userRequestOptions)
+      : await createTemplate(this.entity, this.context.userRequestOptions);
```

Comment:
Similar to above - should we create new Solution items? If we allow it, someone will do it :)